src/parquet/metadata.h

```diff
@@ -38,9 +38,9 @@ using KeyValueMetadata = ::arrow::KeyValueMetadata;
 class ApplicationVersion {
  public:
   // Known Versions with Issues
-  static const ApplicationVersion PARQUET_251_FIXED_VERSION;
-  static const ApplicationVersion PARQUET_816_FIXED_VERSION;
-  static const ApplicationVersion PARQUET_CPP_FIXED_STATS_VERSION;
+  static const ApplicationVersion& PARQUET_251_FIXED_VERSION();
```

Comment:
I don't think you need to break the API anymore by making these functions rather than attributes.

Reply:
We need to keep these as functions to avoid "static initialization order fiasco"

Reply:
What do you mean?

"""In short, suppose you have two static objects x and y which exist in separate source files, say x.cpp and y.cpp. Suppose further that the initialization for the y object (typically the y object’s constructor) calls some method on the x object."""

That doesn't seem to apply here (there are no external static objects involved in the initialization of the application versions). That said, I'm ok with leaving these as methods if you think it's more future-proof.

Reply:
Yeah, my idea is to keep it future-proof.